hips/draw/paint.py

```diff
@@ -109,30 +112,18 @@ def projection(self, tile: HipsTile) -> ProjectiveTransform:
         pt.estimate(src, dst)
         return pt
 
-    def _fetch_tiles(self) -> Iterator[HipsTile]:
-        """Generator function to fetch HiPS tiles from a remote URL."""
-        if self.progress_bar:
-            from tqdm import tqdm
-            tile_indices = tqdm(self.tile_indices, desc='Fetching tiles')
-        else:
-            tile_indices = self.tile_indices
-
-        for healpix_pixel_index in tile_indices:
-            tile_meta = HipsTileMeta(
-                order=self.draw_hips_order,
-                ipix=healpix_pixel_index,
-                frame=self.hips_survey.astropy_frame,
-                file_format=self.tile_format,
-            )
-            url = self.hips_survey.tile_url(tile_meta)
-            tile = HipsTile.fetch(tile_meta, url)
-            yield tile
-
     @property
     def tiles(self) -> List[HipsTile]:
         """List of `~hips.HipsTile` (cached on multiple access)."""
+        tile_fetcher = HipsTileFetcher(self.tile_indices, self.draw_hips_order, self.hips_survey,
+                                       self.tile_format, self.progress_bar)
         if self._tiles is None:
-            self._tiles = list(self._fetch_tiles())
+            if self.fetch_package == 'aiohttp':
```

Comment:
Here in `draw.py` you have the caller of the `HipsTileFetcher`. It should be presented with a uniform and simple interface for both cases. Suggest to pass the option how to fetch to `HipsTileFetcher.__init__` and then run and access the tiles in a uniform way here, without if / else. In practice this means that you move this if / else to a method on the `HipsTileFetcher`.